tests/stdlib/tstrutils.nim

```diff
@@ -221,6 +221,86 @@ template main() =
     doAssert "0123456789ABCDEFGH".find({'A'..'C'}, 5, 10) == 10
     doAssert "0123456789ABCDEFGH".find({'A'..'C'}, 5, 9) == -1
 
+    block:
```

Comment:
why not fix #18128 instead (and adding corresponding tests) ?

Reply:
I definitely plan to fix #18128, but would like these test cases in place first so that the eventual PR for #18128 will include the altered test cases, making it clear how the behaviors of `find`/`rfind` have changed.

Reply:
ok